dao/src/main/kotlin/tables/runs/advisor/AdvisorResultsTable.kt

```diff
@@ -55,15 +55,23 @@ class AdvisorResultDao(id: EntityID<Long>) : LongEntity(id) {
 
     var vulnerabilities by VulnerabilityDao via AdvisorResultsVulnerabilitiesTable
     var defects by DefectDao via AdvisorResultsDefectsTable
-    var issues by IssueDao via AdvisorResultsIssuesTable
 
-    fun mapToModel() = AdvisorResult(
+    fun mapToModel(identifierIssues: List<Issue>) = AdvisorResult(
         advisorName = advisorName,
         capabilities = capabilities.filter(String::isNotEmpty),
         startTime = startTime,
         endTime = endTime,
-        issues = issues.map(IssueDao::mapToModel),
+        issues = filterResultIssues(identifierIssues),
         defects = defects.map(DefectDao::mapToModel),
         vulnerabilities = vulnerabilities.map(VulnerabilityDao::mapToModel)
     )
+
+    /**
+     * Filter the given [identifierIssues] for the issues belonging to this result. This is done based on the
+     * assumption that the source of the issues contains the advisor name. This solution is a bit hacky, but it
+     * should work if the advisors follow default conventions. Note that the primary use case is to fetch the issues
+     * for a whole ORT run and not necessarily drilling this down to single advisor results.
+     */
+    private fun filterResultIssues(identifierIssues: List<Issue>): List<Issue> =
+        identifierIssues.filter { it.source.lowercase().contains(advisorName.lowercase()) }
```

Comment:
Prefer Kotlin's `in` operator over Java's `contains()`.

Reply:
Or use `it.source.contains(advisorName, ignoreCase = true)` to avoid the `lowercase()` calls.

Reply:
Yes, that's even better.